Issue #4628 - Ensuring checkEnabledModules is conditional dependency aware

joakime opened the pull request:
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Issue #4628 - Ensuring checkEnabledModules is required dependency aware

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

gregw reviewed (changes requested):
I don't think this is necessary as the required dependency mechanism duplicates the existing `[optional]` mechanism and should be reverted.

gregw commented:
As explained in #4628, the issue I have with these PRs is that `Required` is the wrong adjective for this mechanism.   `Conditional` is much better, even if currently the only condition we can apply is existence.  Can this PR be used to fix the naming as well as making checkEnabledModules aware of conditional dependencies.

sbordet committed:
Issue #4628 - Add support for conditional module dependencies in jetty-start.

Updated after review.
Renamed isRequiredModule() -> isConditionalModule() and inverted
expressions that were using the method.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet commented:
@gregw I have addressed your naming concern, can you please re-review?

sbordet committed:
Merged branch 'jetty-9.4.x' into 'jetty-9.4.x-4628-start-module-non-required'.
